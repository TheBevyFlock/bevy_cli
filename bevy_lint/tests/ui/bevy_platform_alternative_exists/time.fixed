#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::bevy_platform_alternative_exists)]
#![allow(dead_code)]

use bevy::platform::time::Instant;
//~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
//~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
// type bevy::platform::time::Instant

// Ensure the lint can be muted on imports.
#[allow(bevy::bevy_platform_alternative_exists, unused_imports)]
use std::time::Instant as muted;

struct Player {
    attack_cd: bevy::platform::time::Instant,
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant
}

fn main() {
    let _time = bevy::platform::time::Instant::now();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant

    // This should be fine even tho it will result in a `std::time::Instant` after full path
    // resolution.
    let _bevy_time = bevy::platform::time::Instant::now();

    // Ensure the lint can be muted.
    #[allow(bevy::bevy_platform_alternative_exists)]
    let _time = std::time::Instant::now();

    let _player_bevy = Player {
        attack_cd: bevy::platform::time::Instant::now(),
    };

    let _player = Player {
        attack_cd: bevy::platform::time::Instant::now(),
        //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
        //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
        // type bevy::platform::time::Instant
    };

    let _time = Instant::now();

    let _time: bevy::platform::time::Instant = Instant::now();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant

    // type bevy::platform::time::Instant
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    //~v ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    let _time: bevy::platform::time::Instant = bevy::platform::time::Instant::now();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant

    let _bevy_time: bevy::platform::time::Instant = bevy::platform::time::Instant::now();
}
