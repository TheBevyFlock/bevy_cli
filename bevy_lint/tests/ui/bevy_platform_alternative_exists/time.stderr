error: Used type from the `std` that has an existing alternative from `bevy_platform`
 --> tests/ui/bevy_platform_alternative_exists/time.rs:6:5
  |
6 | use std::time::Instant;
  |     ^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> tests/ui/bevy_platform_alternative_exists/time.rs:3:9
  |
3 | #![deny(bevy::bevy_platform_alternative_exists)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
  |
6 - use std::time::Instant;
6 + use bevy::platform::time::Instant;
  |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/time.rs:16:16
   |
16 |     attack_cd: std::time::Instant,
   |                ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
16 -     attack_cd: std::time::Instant,
16 +     attack_cd: bevy::platform::time::Instant,
   |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/time.rs:23:17
   |
23 |     let _time = std::time::Instant::now();
   |                 ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
23 -     let _time = std::time::Instant::now();
23 +     let _time = bevy::platform::time::Instant::now();
   |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/time.rs:41:20
   |
41 |         attack_cd: std::time::Instant::now(),
   |                    ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
41 -         attack_cd: std::time::Instant::now(),
41 +         attack_cd: bevy::platform::time::Instant::now(),
   |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/time.rs:49:16
   |
49 |     let _time: std::time::Instant = Instant::now();
   |                ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
49 -     let _time: std::time::Instant = Instant::now();
49 +     let _time: bevy::platform::time::Instant = Instant::now();
   |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/time.rs:57:37
   |
57 |     let _time: std::time::Instant = std::time::Instant::now();
   |                                     ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
57 -     let _time: std::time::Instant = std::time::Instant::now();
57 +     let _time: std::time::Instant = bevy::platform::time::Instant::now();
   |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/time.rs:57:16
   |
57 |     let _time: std::time::Instant = std::time::Instant::now();
   |                ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
57 -     let _time: std::time::Instant = std::time::Instant::now();
57 +     let _time: bevy::platform::time::Instant = std::time::Instant::now();
   |

error: aborting due to 7 previous errors

